src/index.js

```diff
@@ -123,8 +123,9 @@ class HCaptcha extends React.Component {
 
       // If they have changed, remove current captcha and render a new one
       if (!match) {
-        this.removeCaptcha();
-        this.renderCaptcha();
+        this.removeCaptcha(() => {
```

Comment:
Looks like this is a race condition, due to the `setState` callback you've accounted for here.